drivers/bq27441/bq27441.c

```diff
@@ -0,0 +1,727 @@
+    return _bq27441_write_extended_data(dev, BQ27441_ID_DISCHARGE, 0, thresholds, 2);
+}
+
+// Get SOCF_Set Threshold - threshold to set the alert flag
+uint8_t bq27441_get_socf_set_threshold(bq27441_t *dev)
+{
+    return _bq27441_read_extended_data(dev, BQ27441_ID_DISCHARGE, 2);
+}
+
+// Get SOCF_Clear Threshold - threshold to clear the alert flag
+uint8_t bq27441_get_socf_clear_threshold(bq27441_t *dev)
+{
+    return _bq27441_read_extended_data(dev, BQ27441_ID_DISCHARGE, 3);
+}
+
+// Set the SOCF set and clear thresholds to a percentage
+bool bq27441_set_socf_thresholds(bq27441_t *dev, uint8_t set, uint8_t clear)
+{
+    uint8_t thresholds[2];
+
+    thresholds[0] = _constrain(set, 0, 100);
+    thresholds[1] = _constrain(clear, 0, 100);
+    return _bq27441_write_extended_data(dev, BQ27441_ID_DISCHARGE, 2, thresholds, 2);
+}
+
+// Check if the SOC1 flag is set
+bool bq27441_get_soc1_flag(const bq27441_t *dev)
+{
+    uint16_t flagState = bq27441_get_flags(dev);
+
+    return flagState & BQ27441_FLAG_SOC1;
+}
+
+// Check if the SOCF flag is set
+bool bq27441_get_socf_flag(const bq27441_t *dev)
+{
+    uint16_t flagState = bq27441_get_flags(dev);
+
+    return flagState & BQ27441_FLAG_SOCF;
+
+}
+
+// Get the SOC_INT interval delta
+uint8_t bq27441_get_soc_int_delta(bq27441_t *dev)
+{
+    return _bq27441_read_extended_data(dev, BQ27441_ID_STATE, 26);
+}
+
+// Set the SOC_INT interval delta to a value between 1 and 100
+bool bq27441_set_soc_int_delta(bq27441_t *dev, uint8_t delta)
+{
+    uint8_t soci = _constrain(delta, 0, 100);
+
+    return _bq27441_write_extended_data(dev, BQ27441_ID_STATE, 26, &soci, 1);
+}
+
+// Pulse the GPOUT pin - must be in SOC_INT mode
+bool bq27441_pulse_gpout(const bq27441_t *dev)
+{
+    return _bq27441_execute_control_word(dev, BQ27441_CONTROL_PULSE_SOC_INT);
+}
+
+// Read the device type - should be 0x0421
+uint16_t bq27441_get_device_type(const bq27441_t *dev)
+{
+    return _bq27441_read_control_word(dev, BQ27441_CONTROL_DEVICE_TYPE);
+}
+
+// Enter configuration mode - set userControl if calling from an Arduino sketch
+// and you want control over when to exitConfig
+bool bq27441_enter_config_mode(bq27441_t *dev, bool userControl)
+{
+    if (userControl) {
+        dev->userConfigControl = true;
+    }
+
+    if (_bq27441_sealed(dev)) {
+        dev->sealFlag = true;
+        _bq27441_unseal(dev); // Must be unsealed before making changes
+    }
+
+    if (_bq27441_execute_control_word(dev, BQ27441_CONTROL_SET_CFGUPDATE)) {
+        int16_t timeout = BQ72441_I2C_TIMEOUT;
+        while ((timeout--) && (!(bq27441_get_status(dev) & BQ27441_FLAG_CFGUPMODE))) {
+            xtimer_nanosleep(1000);
+        }
+
+        if (timeout > 0) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+// Exit configuration mode with the option to perform a resimulation
+bool bq27441_exit_config_mode(const bq27441_t *dev, bool resim)
+{
+    // There are two methods for exiting config mode:
```

Comment:
this is helpful documentation but should be moved to the header file so a developer using the api actually sees it.